src/pages/user/UserInstitution.tsx

```diff
@@ -72,89 +72,121 @@ const UserInstitution: FC = () => {
+        id: values.id,
+        subunits: values.subunits,
+        unit: values.unit,
+      });
+      resetForm(emptyFormikUnit);
+    }
     setOpen(false);
+    setEditMode(false);
   };
 
-  const onSubmit = async (values: FormikInstitutionUnit, { resetForm }: any) => {
-    handleAddInstitution({ name: values.name, id: values.id, subunits: values.subunits });
-    resetForm(emptyFormikUnit);
+  const handleEdit = () => {
+    setOpen(true);
+    setEditMode(true);
   };
 
   return (
     <Card>
       <Heading>{t('heading.organizations')}</Heading>
-      {units.length > 0 ? (
-        units.map((unit: InstitutionUnit, index: number) => <InstitutionCard key={index} unit={unit} />)
-      ) : (
-        <>{!open && <NormalText>{t('organization.no_institutions_found')}</NormalText>}</>
-      )}
       <Formik enableReinitialize initialValues={emptyFormikUnit} onSubmit={onSubmit} validateOnChange={false}>
         {({ values, setFieldValue, handleSubmit, resetForm }: FormikProps<FormikInstitutionUnit>) => (
           <Field name={FormikInstitutionUnitFieldNames.UNIT}>
-            {({ field: { name, value } }: FieldProps) =>
-              open && (
-                <StyledInstitutionSearchContainer>
-                  <InstitutionSearch
-                    dataTestId="autosearch-institution"
-                    label={t('organization.institution')}
-                    clearSearchField={values.name === ''}
-                    setValueFunction={inputValue => {
-                      setFieldValue(FormikInstitutionUnitFieldNames.NAME, inputValue.name);
-                      setFieldValue(FormikInstitutionUnitFieldNames.ID, inputValue.id);
-                      setFieldValue(name, inputValue ?? emptyRecursiveUnit);
-                    }}
-                    placeholder={t('organization.search_for_institution')}
-                  />
-                  {values.unit && (
-                    <>
-                      <InstitutionSelector counter={0} unit={value} />
-                      <StyledButton
-                        onClick={() => handleSubmit()}
-                        variant="contained"
-                        type="submit"
-                        color="primary"
-                        disabled={!values.unit}
-                        data-testid="institution-add-button">
-                        {t('common:add')}
-                      </StyledButton>
-                      <StyledButton
-                        onClick={() => {
-                          toggleOpen();
-                          resetForm({});
-                        }}
-                        variant="contained"
-                        color="secondary">
-                        {t('common:cancel')}
-                      </StyledButton>
-                    </>
-                  )}
-                </StyledInstitutionSearchContainer>
-              )
-            }
+            {({ field: { name, value } }: FieldProps) => (
```

Comment:
ligger mye div inni denne Fielden. ser ut som vi bare trenger field-ting for props til `<InstitutionSearch>` og for å vise valgt verdi, så hva med å flytte resten utenfor? kanskje utenfor `<Formik>` også?
jeg tenker at det som ligger inni en `Field` skal representere selve input-feltet, og minst mulig annet

Reply:
Ja, jeg skjønner hva du mener. Har prøvd å flytte ut deler uten hell selv også. Sånn som det er nå er alt som ligger inni der en del av field'en som er den som heter `unit`. Men jeg synes selv at denne fila er alt for stor, så kan se mer på det i morgen 👍